packages/client/src/encryption/EncryptionUtil.ts

```diff
@@ -108,16 +108,24 @@ export class EncryptionUtil {
             if (newGroupKey) {
                 // newGroupKey should be EncryptedGroupKey | GroupKey, but GroupKey is not defined in protocol
                 // @ts-expect-error expecting EncryptedGroupKey
-                streamMessage.newGroupKey = GroupKey.from([
-                    newGroupKey.groupKeyId,
-                    this.decrypt(newGroupKey.encryptedGroupKeyHex, groupKey)
-                ])
+                streamMessage.newGroupKey = EncryptionUtil.decryptGroupKey(newGroupKey, groupKey)
             }
         } catch (err) {
             streamMessage.encryptionType = StreamMessage.ENCRYPTION_TYPES.AES
             throw new UnableToDecryptError('Could not decrypt new group key: ' + err.stack, streamMessage)
         }
         /* eslint-enable no-param-reassign */
     }
+
+    static encryptGroupKey(nextGroupKey: GroupKey, currentGroupKey: GroupKey): EncryptedGroupKey {
+        return new EncryptedGroupKey(nextGroupKey.id, this.encrypt(nextGroupKey.data, currentGroupKey))
+    }
+
+    static decryptGroupKey(newGroupKey: EncryptedGroupKey, currentGroupKey: GroupKey): GroupKey {
```

Comment:
Could be made `private`?
```suggestion
    private static decryptGroupKey(newGroupKey: EncryptedGroupKey, currentGroupKey: GroupKey): GroupKey {
```